### PR TITLE
sandbox-classic: Add ErrorInfo metadata for rejections. [KVL-1048]

### PR DESCRIPTION
**Pull request by SamirTalwar-DA**
This includes a bunch of extra tests for `SqlLedger`. I enabled the append-only index for these tests as the rejection reason details are not included if we use the mutable index.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/sandbox-classic/src/test/suite/scala/platform/sandbox/stores/ledger/sql/SqlLedgerSpec.scala [hunk 1]
```diff
@@ -273,65 +273,217 @@ final class SqlLedgerSpec
...
+      } yield {
+        result should be(SubmissionResult.Acknowledged)
+        completion._1 should be > start
+        inside(completion._2.completions) {
+          case Seq(Completion(`commandId`, Some(status), _, _, _, _, _)) =>
+            status.code should be(Status.Code.ABORTED.value)
+            val lowerBound = nowInstant.minus(minSkew)
+            val upperBound = nowInstant.plus(maxSkew)
+            status.message should be(
+              s"Ledger time 2021-09-01T18:05:00Z outside of range [2021-09-01T17:59:50Z, 2021-09-01T18:00:30Z]"
```
Comment (miklos-da) [inline ledger/sandbox-classic/src/test/suite/scala/platform/sandbox/stores/ledger/sql/SqlLedgerSpec.scala:434]: Did you want to interpolate the actual lower and upper bound values here?

File: ledger/sandbox-classic/src/test/suite/scala/platform/sandbox/stores/ledger/sql/SqlLedgerSpec.scala [hunk 2]
```diff
@@ -273,65 +273,217 @@ final class SqlLedgerSpec
...
+  private def createSqlLedgerWithParticipantId(participantId: ParticipantId): Future[Ledger] =
+    createSqlLedger(ledgerId = None, participantId = Some(participantId), packages = List.empty)
 
   private def createSqlLedger(
       ledgerId: Option[LedgerId],
       participantId: Option[ParticipantId],
       packages: List[DamlLf.Archive],
-      validatePartyAllocation: Boolean,
+      timeProvider: TimeProvider = TimeProvider.UTC,
+      enableAppendOnlySchema: Boolean = true,
```
Comment (miklos-da) [inline ledger/sandbox-classic/src/test/suite/scala/platform/sandbox/stores/ledger/sql/SqlLedgerSpec.scala:486]: :+1: This is a good default, as we intend to phase out the mutable schema anyway, so eventually all test cases should be using `enableAppendOnlySchema = true`.

**Review by miklos-da — APPROVED** [review]

File: ledger/sandbox-classic/src/test/lib/scala/platform/sandbox/stores/ledger/RejectionSpec.scala [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      Rejection.NoLedgerConfiguration.toStateRejectionReason should be(
+        state.Update.CommandRejected.FinalReason(
+          StatusProto.of(
+            code = Status.Code.ABORTED.value,
+            message = "No ledger configuration available, cannot validate ledger time",
+            details = Seq(
+              AnyProto.pack(
+                ErrorInfo.of(
+                  reason = "NO_LEDGER_CONFIGURATION",
+                  domain = "com.daml.on.sql",
```
[the remark this answers is not in the document]
Comment (SamirTalwar-DA) [reply ledger/sandbox-classic/src/test/lib/scala/platform/sandbox/stores/ledger/RejectionSpec.scala:29]: I'd rather not. I want tests to fail if that constant changes. We should not be changing it on a whim.

File: ledger/sandbox-classic/src/test/suite/scala/platform/sandbox/stores/ledger/sql/SqlLedgerSpec.scala (hunk 1 above)
Comment (SamirTalwar-DA) [reply ledger/sandbox-classic/src/test/suite/scala/platform/sandbox/stores/ledger/sql/SqlLedgerSpec.scala:434]: I did, and then decided that would not be testing anything useful, so I inlined them.

File: ledger/sandbox-classic/src/test/suite/scala/platform/sandbox/stores/ledger/sql/SqlLedgerSpec.scala [hunk 4]
```diff
@@ -273,65 +273,217 @@ final class SqlLedgerSpec
       }
     }
 
-  }
+    "publish a transaction" in {
+      val now = Time.Timestamp.assertFromInstant(Instant.now())
```
[the remark this answers is not in the document]
Comment (SamirTalwar-DA) [reply ledger/sandbox-classic/src/test/suite/scala/platform/sandbox/stores/ledger/sql/SqlLedgerSpec.scala:273]: Good idea. Done.

**Pull request merged**